src/app/group/group-list/group-list.component.ts

```diff
@@ -8,61 +13,131 @@ import { ActivatedRoute, Router } from '@angular/router';
-  constructor( private router: Router, private activatedRoute: ActivatedRoute){};
+  constructor(private router: Router,
+    private activatedRoute: ActivatedRoute,
+    private groupsService: GroupsService,
+    private expenseService: ExpenseService
+  ) {
+    this.getGroupList();
+  };
 
   // ngOnInit method
   ngOnInit(): void {
-    
+
     // get activatedRoute parameter using observable
-    this.activatedRoute.params.subscribe((param) =>{
-      this.getActivatedRouteParam = param['routerParam'];
-      if(this.getActivatedRouteParam === undefined ){
+    this.activatedRoute.params.subscribe((param) => {
+      this.getActivatedRouteParam = param['groupName'];
+      if (this.getActivatedRouteParam === undefined) {
         this.isGroupDetailActive = false;
       }
-      else{
+      else {
         this.isGroupDetailActive = true;
         this.groupNameDetail = this.getActivatedRouteParam;
       }
     })
+    // function call to get total expense
+    this.getExpensesOfGroup();
   }
 
-  // Group detail show
-  groupDetailShow(name: string, index: number): void{
+  /** Group details show function use index and group name from group list
+   * and redirect to group details page.
+   * @param name name of the group
+   * @param index index of the group from grouplist
+   **/
+  groupDetailShow(name: string, index: number): void {
     this.getActivatedRouteParam = name;
-    this.router.navigate(['group','group-list',name]);
+    this.router.navigate(['group', 'group-list', { groupName: name, index: index }]);
+  }
+
+  /** navgate to edit form function use index from group list
+   *  and redirect to edit form with index of group to edit details
+   * @param index index of group from group list
+   **/
+  navigateToEditForm(index: number): void {
+    this.router.navigate(['group', 'group-edit', index]);
   }
 
-  // Navigate to edit form
-  navigateToEditForm(index: number): void{
-    this.router.navigate(['group','group-edit',index]);
+  /**
+   * Set the value of delete expense name and id
+   * @param expenseName - string, expenseId -string
+   * @returns void
+   */
+  setDeleteGroup(groupName: string, groupId: number): void{
+    this.deleteGroupName = groupName;
+    this.deleteGroupId = groupId;
   }
 
-  // Delete group
+  /**  Delete group function
+   * to delete group from list
+   * @param index index of group from list to delete the selected group
+   **/
   deleteGroup(index: number): void {
-    this.groups.splice(index,1);
+    this.groupList.splice(index, 1); // delete row from table
+    const id = this.groupList[index].id;
+    this.groupsService.deleteGroupFromlist(id).subscribe(res => { });
+    this.router.navigate(['group', 'group-list']);
+  }
+
+  /** Group list function to call get api
+   * and get group details from server
+   **/
+  getGroupList(): void {
+    this.groupsService.getGroupList().subscribe(
+      (res) => { this.groupList = res.items },
+      (error) => { alert(error) }
```

Comment:
same as above

Reply:
Sorry di, It was my mistake i have made the changes in local and forgot to push the changes. now it is resolved.